internal/cmd/internal/olmv1/operator_update.go

```diff
@@ -0,0 +1,40 @@
+package olmv1
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/operator-framework/kubectl-operator/internal/cmd/internal/log"
+	v1action "github.com/operator-framework/kubectl-operator/internal/pkg/v1/action"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+// NewOperatorUpdateCmd allows updating a selected operator
+func NewOperatorUpdateCmd(cfg *action.Configuration) *cobra.Command {
+	i := v1action.NewOperatorUpdate(cfg)
+	i.Logf = log.Printf
+
+	cmd := &cobra.Command{
+		Use:   "operator <operator>",
+		Short: "Update an operator",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			i.Package = args[0]
+			_, err := i.Run(cmd.Context())
+			if err != nil {
+				log.Fatalf("failed to update operator: %v", err)
+			}
+			log.Printf("operator %q updated", i.Package)
+		},
+	}
+	bindOperatorUpdateFlags(cmd.Flags(), i)
+
+	return cmd
+}
+
+func bindOperatorUpdateFlags(fs *pflag.FlagSet, i *v1action.OperatorUpdate) {
+	fs.StringVar(&i.Version, "version", "", "desired operator version (single or range) in semver format. AND operation with channels")
+	fs.StringArrayVar(&i.Channels, "channels", []string{}, "desired channels for operator versions. AND operation with version")
+	fs.StringVar(&i.UpgradeConstraintPolicy, "upgrade-constraint-policy", "", "controls whether the upgrade path(s) defined in the catalog are enforced, one of CatalogProvided|SelfCertified), Default: CatalogProvided")
```

Comment:
```suggestion
fs.StringVar(&i.UpgradeConstraintPolicy, "upgrade-constraint-policy", "", 
    "controls whether the upgrade path(s) defined in the catalog are enforced. OPTIONS: 'CatalogProvided', 'SelfCertified'. Default: 'CatalogProvided'.")



```

Reply:
with `one of:` I was following what olmv0 used in some of their commands, I think we can revisit that when we dive deeper into docs in the (near) future, context: https://github.com/operator-framework/kubectl-operator/pull/225#discussion_r1984908346

Reply:
When we review a PR, we spend time analyzing the code, identifying areas for improvement, and raising necessary changes. Ensuring that help information is clear and useful is part of this process. I can share many PRs where we’ve done this as an example.

Before this PR gets merged, I believe we should ensure that the command help text is clean and well-structured.

So, can we keep it open as a TODO before merging?

Reply:
Can we please try to use a more common ux friendly description, similar to other tools, operator-controller flags, etc. before get the PR merged?

Reply:
just for the record, you're proposing to change:
```
One of CatalogProvided|SelfCertified), Default: CatalogProvided"
```

to:
```
OPTIONS: 'CatalogProvided', 'SelfCertified'. Default: 'CatalogProvided'."
```

I am sorry, but I'd rather follow olmv0 pattern (meaning `One of, |`) than an arbitrary and unspecified `common ux friendly description`. I'd ask you to respect that and not unresolve the comment - if you feel strongly about this, let's revisit when we'll be reviewing the documentation as a whole before we get to RC with those new commands (https://github.com/operator-framework/kubectl-operator/pull/225#issuecomment-2710087085).
Hope that's fair.

Reply:
Yes, that is my suggestion.
I think that is how we have it controller-tools as well, but it does not seem very significant as well and we have it in OLMv0 I am fine,